text/0000-reactive-assignments.md

```diff
@@ -607,9 +607,46 @@ peekaboo.show();
 An `export const` would be a signal to the compiler that a property is read-only — in other words, attempting to assign to it would cause an error.
 
 
-### Preload
+### `preload` and `setup`
+
+There is a rather awkward mechanism for declaring static properties on a component constructor in Svelte 2 — the `setup` hook:
+
+```html
+<script>
+  export default {
+    setup(Component) {
+      Component.staticMethod = () => {...};
+    }
+  };
+</script>
+```
+
+This is deeply weird, and due to an oversight (that we can't correct without a breaking change) only runs on the client.
+
+Since Sapper requires that components have some way of declaring their data dependencies prior to rendering, and since `setup` is so cumbersome, there is a special case made for `preload`. The `preload` function is attached to components on both client and server, and has no well-defined behaviour; it is purely convention.
+
+We can do better, **but it requires something potentially controversial** — a second `<script>` block that, unlike the one we've been using so far, runs a single time rather than upon every instantiation. Let's call it (🐃) `scope="shared"`:
+
+```html
+<script scope="shared">
```

Comment:
As mentioned in chat, I would _really_ like to see this be written as this instead:

```html
<!-- ambiguous "preload" applies to both scopes -->
<script preload>
  export function preload({ params }) {
    //
  }
</script>

<!-- specify "preload" scope -->
<script preload="client|server">
  export function preload({ params }) {
    //
  }
</script>
```

Reply:
I actually think the unification of the different types of static stuff that can be attached to a component is nice - and being able to make `preload` just be a convention is great. (If we want to have different client/server preloads, could that just be part of Sapper's convention?)

That said, I am really not fond of the term `scope='shared'` that's being proposed to indicate this. I don't have a better suggestion currently.

Reply:
I'm inclined to agree with @Conduitry — `preload` is an awkward anomaly. The less we have those special cases the more stable a platform for innovation the framework becomes

Reply:
TBH I'm not sure what either of you are saying 😅

Reply:
There's nothing special about `preload` — it doesn't have any defined behaviour. It's just a function that will get attached to both client and server components as a static method, and it gets that privileged treatment because Sapper needs it.

But ideally Svelte shouldn't have features that exist just for the sake of Sapper — someone with different opinions about how to build an app framework with Svelte should be able to do so. 

This RFC, among other things, is about stripping Svelte down to its bare minimum — two methods instead of five, one way of making values available to the markup instead of (3? 10? depends which ones you count).

Since we need some way of including code that runs once per app (instead of once per component), we need to have the second `<script>` tag. Since we have that second script tag, we may as well use it for `preload` rather than it being its own special thing.

Reply:
100% agreed - apologies if something I said pointed to the contrary. I was merely just renaming `scope` 

Having extra script tags also _allows_ new app frameworks to define new boundaries for new behaviors, especially if/when Svelte users are accustomed to having them already.

Reply:
Ah ok, I thought you were implying that the second `<script>` was purely for `preload`. Certainly in the market for alternatives to `scope`

Reply:
No, but in Sapper apps that would/could be the place for it... Standlone, hoistable preload functions that the Svelte compiler can rip out (safely) per-bundle-type and put those snippets wherever Sapper tells it to put them.

Svelte just needs to be aware of/handle more than 1 script tag